Unable to create a block custom volume

LXD supports 2 types of ["custom" storage volumes](https://linuxcontainers.org/lxd/docs/master/howto/storage_volumes/#create-the-volume): `filesystem` and `block`.

When creating a volume using the [lxd_volume](https://registry.terraform.io/providers/terraform-lxd/lxd/latest/docs/resources/volume) resource of the provider, we can specify the `type = "custom"` but we cannot choose the volume `content_type` which defaults to `filesystem`.

We should be able to specify the `content_type` to either `filesystem` or `block`.

I tried adding this capability, but the lxd api dependency is years out of date and does not support `content_type`. 

```
lxd/resource_lxd_volume.go:103:18: invalid operation: operator + not defined on d.Get("content_type").(string) (comma, ok 
expression of type string)
lxd/resource_lxd_volume.go:109:10: volume.ContentType undefined (type api.StorageVolumesPost has no field or method Conten
tType)
lxd/resource_lxd_volume.go:109:24: undefined: contentType
```

--- a/docs/resources/volume.md
+++ b/docs/resources/volume.md
@@ -39,6 +39,8 @@ resource "lxd_volume" "volume1" {
 
 * `project` - *Optional* - Name of the project where the volume will be stored.
 
+* `content_type` - *Optional* - Volume content type (filesystem or block)
+
 ## Attribute Reference
 
 The following attributes are exported:

--- a/lxd/resource_lxd_volume.go
+++ b/lxd/resource_lxd_volume.go
@@ -69,6 +69,13 @@ func resourceLxdVolume() *schema.Resource {
 				Optional: true,
 				ForceNew: true,
 			},
+
+			"content_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "filesystem",
+			},
 		},
 	}
 }
@@ -94,12 +101,14 @@ func resourceLxdVolumeCreate(d *schema.ResourceData, meta interface{}) error {
 	pool := d.Get("pool").(string)
 	volType := d.Get("type").(string)
 	config := resourceLxdConfigMap(d.Get("config"))
+	content_type := d.Get("content_type").(string)
 
 	log.Printf("Attempting to create volume %s", name)
 	volume := api.StorageVolumesPost{}
 	volume.Name = name
 	volume.Type = volType
 	volume.Config = config
+	volume.ContentType = content_type
 	if err := server.CreateStoragePoolVolume(pool, volume); err != nil {
 		return err
 	}

--- a/lxd/resource_lxd_volume_test.go
+++ b/lxd/resource_lxd_volume_test.go
@@ -175,6 +175,27 @@ func testAccVolumeConfig(volume *api.StorageVolume, k, v string) resource.TestCh
 	}
 }
 
+func TestAccVolume_contentType(t *testing.T) {
+	var volume api.StorageVolume
+	poolName := strings.ToLower(petname.Generate(2, "-"))
+	volumeName := strings.ToLower(petname.Generate(2, "-"))
+	source := t.TempDir()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVolume_contentType(poolName, source, volumeName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccVolumeExists(t, "lxd_volume.volume1", &volume),
+					resource.TestCheckResourceAttr("lxd_volume.volume1", "name", volumeName),
+				),
+			},
+		},
+	})
+}
+
 func testAccVolume_basic(poolName, source, volumeName string) string {
 	return fmt.Sprintf(`
 resource "lxd_storage_pool" "pool1" {
@@ -254,4 +275,22 @@ resource "lxd_volume" "volume1" {
   project = lxd_project.project1.name
 }
 	`, project, volumeName)
+}
+
+func testAccVolume_contentType(poolName, source, volumeName string) string {
+	return fmt.Sprintf(`
+resource "lxd_storage_pool" "pool1" {
+ 	name   = "%s"
+ 	driver = "dir"
+ 	config = {
+ 		source = "%s"
+ 	}
+}
+
+resource "lxd_volume" "volume1" {
+ 	name         = "%s"
+ 	pool         = "${lxd_storage_pool.pool1.name}"
+ 	content_type = "block"
+}
+	`, poolName, source, volumeName)
 }

